signac/contrib/job.py

```diff
@@ -336,12 +409,12 @@ def statepoint(self):
             Returns the job's state point.
 
         """
-        if self._statepoint is None:
-            # Load state point manifest lazily and assign to
-            # self._statepoint
-            statepoint = self._check_manifest()
+        if self._statepoint_requires_init:
+            statepoint = self._statepoint.load(self.id)
```

Comment:
I think this comment is important for developers / readers. It was removed in the diff.
```suggestion
            # Load state point data lazily (on access).
            statepoint = self._statepoint.load(self.id)
```